graphblas/core/ss/descriptor.py

```diff
@@ -13,40 +16,172 @@
 }
 
 
-def set_nthreads(desc, nthreads):
-    check_status(
-        lib.GxB_Desc_set(
-            desc._carg,
-            lib.GxB_NTHREADS,
-            vararg(ffi.cast("int", nthreads)),
-        ),
-        desc,
-    )
+# It would be great if we could make Descriptor class more like this
+class _DescriptorConfig(BaseConfig):
+    _get_function = lib.GxB_Desc_get
+    _set_function = lib.GxB_Desc_set
+    _options = {
+        # GrB
+        "output_replace": (lib.GrB_OUTP, "GrB_Desc_Value"),
+        "mask_complement": (lib.GrB_MASK, "GrB_Desc_Value"),
+        "mask_structure": (lib.GrB_MASK, "GrB_Desc_Value"),
+        "transpose_first": (lib.GrB_INP0, "GrB_Desc_Value"),
+        "transpose_second": (lib.GrB_INP1, "GrB_Desc_Value"),
+        # GxB
+        "nthreads": (lib.GxB_DESCRIPTOR_NTHREADS, "int"),
+        "chunk": (lib.GxB_DESCRIPTOR_CHUNK, "double"),
+        "axb_method": (lib.GxB_AxB_METHOD, "GrB_Desc_Value"),
+        "sort": (lib.GxB_SORT, "int"),
+        "secure_import": (lib.GxB_IMPORT, "int"),
+        # "gpu_control": (GxB_DESCRIPTOR_GPU_CONTROL, "GrB_Desc_Value"),  # Coming soon...
+    }
+    _enumerations = {
+        # GrB
+        "output_replace": {
+            True: True,  # lib.GrB_REPLACE == 1
+            False: False,
+        },
+        "mask_complement": {
+            lib.GrB_COMP: True,
+            False: False,
+        },
+        "mask_structure": {
+            lib.GrB_STRUCTURE: True,
+            False: False,
+        },
+        "transpose_first": {
+            lib.GrB_TRAN: True,
+            False: False,
+        },
+        "transpose_second": {
+            lib.GrB_TRAN: True,
+            False: False,
+        },
+        # GxB
+        "axb_method": {
+            lib.GxB_AxB_GUSTAVSON: "gustavson",
+            lib.GxB_AxB_DOT: "dot",
+            lib.GxB_AxB_HASH: "hash",
+            lib.GxB_AxB_SAXPY: "saxpy",
+            lib.GxB_DEFAULT: "default",
+        },
```

Comment:
How does this mapping work? It feels backwards to me.

> if key in self._enumerations and isinstance(val, str):
            val = val.lower()
            val = self._enumerations[key][val]

Shouldn't the string (ex. "gustavson") be the key and `lib.GxB_AxB_GUSTAVSON` be the value?

Reply:
In `BaseConfig.__init__`, we add the reverse mappings to `self._enumerations`, so you can use either `"gustavson"` or `lib.GxB_AxB_GUSTAVSON`. I'm not sure why `_enumerations` and `_bitwise` map the lib object to string, which feels backwards to me. I guess a possible advantage of doing so is ensuring a 1-to-1 mapping of lib to string, but I don't know why this would be important. Also, having both directions of mappings in the same dict feels weird too.

So, I'm happy to clean this up.

Reply:
Updated. There are now forward and reverse mappings.